invenio_notifications/backends/utils/loaders.py

```diff
@@ -24,8 +25,8 @@ def render_template(self, notification, recipient):
         More specific templates take precedence over less specific ones.
         Rendered template will also take the locale into account.
         """
-        # Take locale into account
-        locale = recipient.data.get("locale", "en")
+        # Take recipient locale into account. Fallback to default locale (set via config variable)
+        locale = recipient.data.preferences.get("locale", get_locale())
```

Comment:
```suggestion
        locale = recipient.data.preferences.get("locale") or get_locale()
```

minor: in case `preferences.locale: None` (not sure if it's even possible, but better be safe than sorry :) ).

Reply:
The `get` method will do the same internally, if a default parameter is specified ()
With the finding of the above comment, this probably should be adapted to something like
```suggestion
        locales = [] 
        if "invenio-i18n" in current_app.extensions:
            locales = [x[0] for x in current_app.extensions["invenio-i18n"].get_languages()]

        locale = recipient.data.get("preferences", {}).get("locale")
        locale = locale if locale in locales else get_locale()            
```

Reply:
I see actually two options for handling this:

1. Checking via `get_languages()` as you suggested.
  - I would prefer though if we could explicitly call some sort of dedicated locale validation function utility... Maybe there's already some util function coming from `flask_babel`?
2. Wrapping the rendering in a `try/except` and "recovering" using the default locale to render

I prefer `1.`, especially if we could also find something from Flask-Babel to do the validation.

Reply:
Investigated a bit and found out that `Locale.parse(locale)` will check if a translation file for the given locale is available. If not, it will raise an `UnknownLocaleError`. However, the configured locales may differ from the ones available specified via the `I18N_LANGUAGES` config variable, so we will have to check that explicitly.
With the proxy added and the additional method, it should ease checking the availability of a locale.

[i18n implementation](https://github.com/inveniosoftware/invenio-i18n/pull/121/files#diff-264718ab6e4dc895921f711fad6c9f48653f8fbe8539ad675aae67090284d9a2R177-R187)